### PR TITLE
- fixes  vishvananda/netlink#815

### LINKED ISSUE #815: xfrm:  v4 over v6 IPsec and v6 over v4 tunnel don't work
currently when creating v4 over v6 tunnel, there are two issues to prevent it from working:

1. the IPv6 tmpl src/dst address of the xfrm policy will be incorrectly treated as IPv4 address 
2. the selector src/dst of the xfrm state is always ::/0, while it should be IPv4

similar things happen with v6 over v4 tunnel

### PR DESCRIPTION
**Pull request by hujun-open**
Fixes  #815 

- changes: 1. set userTmpl.Family to correct family 2. add Selector to XfrmState which is corresponding to XfrmUsersaInfo.Sel

### CODE REVIEW

**Comment by hujun-open** [comment]
This fixes issue #815

File: xfrm_state_linux.go [hunk 1]
```diff
@@ -515,6 +515,10 @@ func xfrmUsersaInfoFromXfrmState(state *XfrmState) *nl.XfrmUsersaInfo {
 	msg.Id.Spi = nl.Swap32(uint32(state.Spi))
 	msg.Reqid = uint32(state.Reqid)
 	msg.ReplayWindow = uint8(state.ReplayWindow)
-
+	if state.Selector != nil {
+		selFromPolicy(&msg.Sel, state.Selector)
+	} else {
+		msg.Sel = nl.XfrmSelector{}
+	}
```
Comment (aboch) [inline xfrm_state_linux.go:522]: for readability, can you 
```
msg.Sel = nl.XfrmSelector{}
if state.Selector != nil {
	selFromPolicy(&msg.Sel, state.Selector)
}
```
instead?

**Comment by aboch** [comment]
is it possible to extend existing UT function to cover this?

**Comment by hujun-open** [comment]
> is it possible to extend existing UT function to cover this?

don't know what does "UT" means, could you point to the specific function?

**Comment by aboch** [comment]
> > is it possible to extend existing UT function to cover this?
> 
> don't know what does "UT" means, could you point to the specific function?

Unit Test
I believe you would add it [here](https://github.com/vishvananda/netlink/blob/main/xfrm_state_test.go) or [here](https://github.com/vishvananda/netlink/blob/main/xfrm_policy_test.go) for your change

**Comment by hujun-open** [comment]
ok, updated based on above discussion

**Comment by aboch** [comment]
Thanks @hujun-open 
Please squash the two commits into one.

**Comment by hujun-open** [comment]
squashed 

> Thanks @hujun-open Please squash the two commits into one.

**Comment by aboch** [comment]
LGTM

**Pull request merged**